pandas/core/groupby/grouper.py

```diff
@@ -387,12 +386,6 @@ def _set_grouper(self, obj: NDFrame, sort: bool = False) -> None:
         self.obj = obj  # type: ignore[assignment]
         self._gpr_index = ax
 
-    @final
```

Comment:
Ah approved to quickly. I have a question:

```
df = DataFrame({"a": [1, 2, 3], "b": 1})
grper = pd.Grouper(key="a")
x = df.groupby(grper)
grper.groups
```

This returns the groups right now and wouldn't work anymore?

Reply:
For me this raises AttributeError on main

Reply:
Ah I wasn't on the newest commit, this has changed in 385a6679d40a359ac93e9c81aef467bc561c1aef

Was this intentional? Should add a whatsnew then, since this is technically user facing

Reply:
oh your right, darn.  i guess will have to revert part of #51145 and do that+this as a deprecation